pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -48,80 +46,109 @@ type (
 )
 
 const (
-	controllerMode = "controller"
-	agentMode      = "agent"
+	AgentMode            = "agent"
+	AgentWindowsMode     = "agent-windows"
+	ControllerMode       = "controller"
+	agentConfigName      = "antrea-agent.conf"
+	controllerConfigName = "antrea-controller.conf"
 )
 
 // HandleFunc returns the function which can handle queries issued by 'antctl get featuregates' command.
 // The handler function populates Antrea featuregates information to the response.
-// For now, it will return all feature gates included in features.DefaultAntreaFeatureGates for agent
-// We need to exclude any new feature gates which is not consumed by agent in the future.
 func HandleFunc(k8sclient clientset.Interface) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		antreaConfigName := env.GetAntreaConfigMapName()
-		antreaConfig, err := k8sclient.CoreV1().ConfigMaps(env.GetAntreaNamespace()).Get(context.TODO(), antreaConfigName, metav1.GetOptions{})
+		antreaConfigMapName := env.GetAntreaConfigMapName()
+		antreaNamespace := env.GetAntreaNamespace()
+		antreaConfig, err := k8sclient.CoreV1().ConfigMaps(antreaNamespace).Get(context.TODO(), antreaConfigMapName, metav1.GetOptions{})
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			klog.Errorf("Error when getting config map %s: %v", antreaConfigName, err)
+			klog.ErrorS(err, "Error when getting config map", "ConfigMap", klog.KRef(antreaNamespace, antreaConfigMapName))
 			return
 		}
+		configMaps, err := k8sclient.CoreV1().ConfigMaps(antreaNamespace).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "app=antrea",
+		})
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			klog.ErrorS(err, "Error when getting antrea-windows config map")
```

Comment:
The error message is not accurate here.
```suggestion
			klog.ErrorS(err, "Error when listing all Antrea ConfigMaps by label")
```